python/plugins/processing/algs/grass7/Grass7Algorithm.py

```diff
@@ -839,28 +862,34 @@ def exportVectorLayerFromParameter(self, name, parameters, context):
                 if layerType in self.QGIS_OUTPUT_TYPES:
                     dataType = self.QGIS_OUTPUT_TYPES[layerType]
 
-        grassName = '{}{}'.format(name, self.uniqueSuffix)
-        self.exportVectorLayer(grassName, fileName, dataType)
+        outFormat = QgsVectorFileWriter.driverForExtension(os.path.splitext(fileName)[1])
+        dsco = self.parameterAsString(parameters, self.GRASS_VECTOR_DSCO, context)
+        lco = self.parameterAsString(parameters, self.GRASS_VECTOR_LCO, context)
+        self.exportVectorLayer(grassName, fileName, layer, nocats, dataType, outFormat, dsco, lco)
 
-    def exportVectorLayer(self, grassName, fileName, dataType='auto', layer=None, nocats=False):
+    def exportVectorLayer(self, grassName, fileName, layer=None, nocats=False, dataType='auto',
+                          outFormat='ESRI_Shapefile', dsco=None, lco=None):
         """
         Creates a dedicated command to export a vector from
-        temporary GRASS DB into a file via ogr.
-        :param grassName: name of the parameter.
-        :param fileName: file path of raster layer.
-        :param dataType: GRASS data type for exporting data.
-        :param layer: In GRASS a vector can have multiple layers.
-        :param nocats: Also export features without category if True.
+        temporary GRASS DB into a file via OGR.
+        :param grassName: name of the vector to export.
+        :param fileName: file path of vector layer.
+        :param dataType: export only this type of data.
+        :param layer: for vector with multiples layers, exports only one layer.
+        :param nocats: do not export GRASS categories.
+        :param outFormat: file format for export.
+        :param dsco: datasource creation options for format.
+        :param lco: layer creation options for format.
         """
         for cmd in [self.commands, self.outputCommands]:
             cmd.append(
-                'v.out.ogr{0} type={1} {2} input="{3}" output="{4}" {5}'.format(
-                    ' -c' if nocats else '',
-                    dataType,
+                'v.out.ogr{0} type="{1}" input="{2}" output="{3}" format="{4}" {5}{6}{7} --overwrite'.format(
+                    '' if nocats else ' -c',
```

Comment:
-c flag: the v.out.ogr manual states: "To export polygons with holes into, e.g., a Esri Shapefile, and make the holes appear as filled areas, the flag -c has to be used." - perhaps no longer desired when switching to GPKG?

Reply:
Hmm, looking at https://grass.osgeo.org/grass74/manuals/v.out.ogr.html seems `-c` flag does different thing: "Also export features without category (not labeled). Otherwise only features with category are exported."